### DISCUSSION TITLE
Compatibility with Page Moderation

### DISCUSSION THREAD

**Discussion by ho-tsl** [wagtail/wagtail-localize#788] [General]
Wagtail has a very useful feature called "Moderation". This feature can be easily activated by creating a Moderation workflow through the admin.
However, the Wagtail localize packages don't have compatibility with this feature. So we cannot send any translated pages to be reviewed by our Editors.
I was wondering how much it would cost to make it compatible?
At least, you could make the React component "TranslationEditor" compatible with a Python setting constant such as BASE_PAGE_ACTION_MENU_ITEMS so we could add our own menus. What do you think?

**Comment by zerolab** [comment]
The first step would be to remove the React side of things. I believe it is now more of a crutch than anything.

Following that, we need to look at what it means for:
- sync translation
- sync translation, and publish

both from a simple workflow point of view (the translation is awaiting moderation, and we're updating essentially creating a new draft. Should the old workflow be stopped and the a new one be created at the same step?) and a more complex one (translation is in multiple steps, and locked for most people)

What impact and considerations are there if the person who is sending the original page for translation does not have permissions in any of the steps in the workflow?

etc.

**Reply by ho-tsl** [reply]
I see the problem.
What if we create a rule that checks if the translated pages are in the middle of a workflow? If so, we totally block the sync translation feature.
I would also block it in the TranslationSource.update_or_create_from_instance() method.
This way, the author, who has no permission in any workflow step, will have only two options to sync translation:
1. Wait until the translated page is approved.
2. Cancel the translated page moderation.

What do you think?

**Comment by ho-tsl** [comment]
Hey @zerolab, I'm building the workflow compatibility here. I can open a PR here afterwards, but there is a thing that's impossible to handle in a proper way.
The following function kidnaps the default page view, locking up all default actions (such as action-submit, the one that sends a page to be moderated).
![image](https://github.com/wagtail/wagtail-localize/assets/158229836/ce9cb67f-14f5-4613-befe-8325b01ddb49)

I was wondering if you could add a "return" at line 249. So if it's a POST request and it is not coming from the wagtail-localize view, it would still work. Is that possible?

**Reply by zerolab** [reply]
In theory it should work. Give it a go and see how it behaves
